shap/_explanation.py

```diff
@@ -1,6 +1,7 @@
 import copy
 import operator
-from typing import Any, Union
+from dataclasses import dataclass, field
+from typing import Any, Union, cast
```

Comment:
Following the addition of the "future" imports in #3754 (which has led to a bit of a merge conflict), we should now be able to avoid having to import `Union` or `Optional`.